File: app/src/main/java/idrisadetunmbi/githubusers/presenters/GithubUsersPresenter.java
package idrisadetunmbi.githubusers.presenters;

import java.util.List;

import idrisadetunmbi.githubusers.UsersContract;
import idrisadetunmbi.githubusers.models.GithubUser;

public class GithubUsersPresenter implements UsersContract.ActionsListener {
    private UsersContract.View mGithubUserView;
    private UsersContract.Data mData;

    public GithubUsersPresenter(UsersContract.View githubUserView, UsersContract.Data data) {
        mGithubUserView = githubUserView;
        mData = data;
    }

    @Override
    public void loadUsers() {
        if (mGithubUserView.userIsOnline()) {
            mGithubUserView.setProgressIndicator(true);
            mData.getUsers(new UsersContract.Data.Callback<List<GithubUser>>() {
                @Override
                public void onLoaded(List<GithubUser> data) {
                    mGithubUserView.setProgressIndicator(false);
                    mGithubUserView.showUsers(data);
                }
            });
        } else {
            mGithubUserView.setProgressIndicator(false);
            mGithubUserView.showNetworkUnavailabilitySnackbar();
        }
    }

    @Override
    public void openDetailsView(GithubUser user) {
        mGithubUserView.showUserDetailsUi(user);
    }
}

File: app/src/main/java/idrisadetunmbi/githubusers/presenters/GithubUsersPresenter.kt
package idrisadetunmbi.githubusers.presenters

import idrisadetunmbi.githubusers.UsersContract
import idrisadetunmbi.githubusers.models.GithubUser

class GithubUsersPresenter(
        private val mGithubUserView: UsersContract.View,
        private val mData: UsersContract.Data) : UsersContract.ActionsListener {

    override fun loadUsers() {
        if (mGithubUserView.deviceIsConnected()) {
            mGithubUserView.setProgressIndicator(true)
            mData.getUsers(object : UsersContract.Data.Callback<List<GithubUser>> {
                override fun onLoaded(data: List<GithubUser>) {
                    mGithubUserView.setProgressIndicator(false)
                    mGithubUserView.showUsers(data)
                }
            })
        } else {
            mGithubUserView.setProgressIndicator(false)
            mGithubUserView.showNetworkUnavailabilitySnackbar()
        }
    }

    override fun openDetailsView(user: GithubUser) {
        mGithubUserView.showUserDetailsUi(user)
    }
}
